internal/filegreeter/greeter.go

```diff
@@ -0,0 +1,63 @@
+package filegreeter
+
+import (
+	"bufio"
+	"context"
+	"github.com/Tommy647/go_example/internal/greeter"
+	"io"
+	"log"
+	"os"
+)
+
+// New returns a new instance of our file greeter
+func New(path string) *FileGreeter {
+	file, err := os.Open(path)
+	defer file.Close()
```

Comment:
defer after the error check, if we erred we don't need to close, and it's easier to read if the `if err != nil` block is kept with the declaration

Reply:
makes sense, cheers